src/server/run_mir.cpp

```diff
@@ -102,6 +102,19 @@ void mir::run_mir(
 
     static std::atomic<unsigned int> concurrent_calls{0};
 
+    if (!concurrent_calls)
```

Comment:
Why is this guarded by `concurrent_calls`? It doesn't guarantee that it's only called once, because it happens before the increment of `concurrent_calls`. Is there any reason not to just do this unconditionally?